plugins/inputs/jenkins/jenkins.go

```diff
@@ -0,0 +1,265 @@
+	fields := make(map[string]interface{})
+
+	// get node data
+	for _, node := range nodes {
+		info, _ := node.Info()
+		tags["node_name"] = node.GetName()
+
+		// skip nodes without information
+		if info.MonitorData.Hudson_NodeMonitors_ArchitectureMonitor == nil {
+			continue
+		}
+
+		tags["arch"] = info.MonitorData.Hudson_NodeMonitors_ArchitectureMonitor.(string)
+		isOnline, _ := node.IsOnline()
+		fields["online"] = int(0)
+		if isOnline {
+			fields["online"] = int(1)
+		}
+
+		fields["response_time"] = info.MonitorData.Hudson_NodeMonitors_ResponseTimeMonitor.Average
+
+		if info.MonitorData.Hudson_NodeMonitors_DiskSpaceMonitor != nil {
+			diskSpace := info.MonitorData.Hudson_NodeMonitors_DiskSpaceMonitor.(map[string]interface{})
+			tags["disk_path"] = diskSpace["path"].(string)
+			fields["disk_available"] = diskSpace["size"].(float64)
+		}
+
+		if info.MonitorData.Hudson_NodeMonitors_TemporarySpaceMonitor != nil {
+			tempSpace := info.MonitorData.Hudson_NodeMonitors_TemporarySpaceMonitor.(map[string]interface{})
+			tags["temp_path"] = tempSpace["path"].(string)
+			fields["temp_available"] = tempSpace["size"].(float64)
+		}
+
+		if info.MonitorData.Hudson_NodeMonitors_SwapSpaceMonitor != nil {
+			swapSpace := info.MonitorData.Hudson_NodeMonitors_SwapSpaceMonitor.(map[string]interface{})
+			fields["swap_available"] = swapSpace["availableSwapSpace"].(float64)
+			fields["swap_total"] = swapSpace["totalSwapSpace"].(float64)
+			fields["memory_available"] = swapSpace["availablePhysicalMemory"].(float64)
+			fields["memory_total"] = swapSpace["totalPhysicalMemory"].(float64)
+		}
+
+		acc.AddFields(measurement, fields, tags)
+
+	}
+
+	return nil
+}
+
+func (j *Jenkins) GetJobsData(jobs []*gojenkins.Job, acc telegraf.Accumulator) error {
+
+	measurement := "jenkins_job"
+
+	for _, job := range jobs {
+
+		jobName := job.GetName()
+
+		for _, filtername := range j.JobFilterName {
+			if filtername == jobName {
+				continue
+			}
+		}
+
+		// ignore if job has no builds
+		if job.Raw.LastBuild.Number < 1 {
+			continue
+		}
+
+		jobLastBuild, err := job.GetLastBuild()
+		if err != nil {
+			log.Printf("E! Error retrieving last build from job %s: %s", jobName, err.Error())
+			return err
+		}
+
+		// ignore if last build is too old
+		if (j.LastbuildFilterInterval != internal.Duration{Duration: 0}) {
+			buildSecAgo := time.Now().Sub(jobLastBuild.GetTimestamp()).Seconds()
+
+			if buildSecAgo > j.LastbuildFilterInterval.Duration.Seconds() {
+				log.Printf("D! Last job too old, last %s build was %v seconds ago", jobName, buildSecAgo)
+				continue
+			}
+		}
+
+		buildIds, err := job.GetAllBuildIds()
+		sort.Sort(byBuildNumber(buildIds))
+
+		if err != nil {
+			log.Printf("E! Error retrieving all builds from job \"%s\": %s", jobName, err.Error())
+			return err
+		}
+
+		for _, buildId := range buildIds {
+			build, err := job.GetBuild(buildId.Number)
+			log.Printf("D! Reading data from job %s build %v", jobName, buildId.Number)
```

Comment:
I would remove this log message, if the build does not result in a metric we will get another log message so I think we are able to debug in either case.